### PR TITLE
introduced check for fatal errors on the connection prior to calling …

### PR DESCRIPTION
**Pull request by RodneyLaver**
…SSL_shutdown

### CODE REVIEW

File: src/tls_openssl.c [hunk 1]
```diff
@@ -201,6 +201,11 @@ int tls_stop(tls_t *tls)
     int error;
     int ret;
 
+    /* According to OpenSSL.org, we must not call SSL_shutdown(3)
+       if a previous fatal error has occurred on a connection. */
+    if (tls->lasterror == SSL_ERROR_SYSCALL || tls->lasterror == SSL_ERROR_SSL)
+    return 1;
```
Comment (pasis) [inline src/tls_openssl.c]: Please, add 4 spaces before the return statement. This is the block of `if` statement and should be indented accordingly.

**Comment by RodneyLaver** [comment]
thanks again for the hints/correction - this is good training - hope the commit it is alright.

File: src/tls_openssl.c [hunk 2]
```diff
@@ -201,6 +201,13 @@ int tls_stop(tls_t *tls)
     int error;
     int ret;
 
+    /* tls/openssl: don't call SSL_shutdown() after a fatal error
```
Comment (pasis) [inline src/tls_openssl.c]: I meant that this message should be in the commit message. That one you write when you run `git commit`. The previous comment here was fine.

The reason I suggested to re-write commit message is because many tools work better with a short 1st line. For example, github will cut a long line. Besides, `git log` is very powerful command when you investigate a problem with code and good commit messages save time.

Also, I would like you to update your 1st commit instead of creating 2nd one. It not only would make git log more tidy, but also allows to use commands `git log|blame|show` efficiently.

So, how you can easily put all changes into 1 commit:
```
# Note that you reset commits till the commit with this hash
git reset c40b4c12c1f81b7c7b4b673cbf19a4ecb4f0c6ab
git add src/tls_openssl.c
git commit
git push -f origin check-for-fatal-error-before-SSL_shutdown
```

If you needed to update only the last commit, you would need to use `git commit --amend` command. This is very useful command if you want to create tidy git logs :).

In conclusion, write the previous comment here, it was fine, make a single commit instead of 2 and move current message to the commit message. If you have any questions, please ask.

**Comment by RodneyLaver** [comment]
hope this is ok now - any further misunderstandig from my side - let me know!!!

**Comment by pasis** [comment]
It is almost perfect now, except of 1 thing: remove `/*` and `*/` in the commit message. You used them for comment in C code, but shouldn't use in commit message. Simply run `git commit --amend` in your branch, change commit message and then force push your commit again `git push -f origin check-for-fatal-error-before-SSL_shutdown`

**Comment by RodneyLaver** [comment]
Ok - so I hope this time...

**Comment by pasis** [comment]
Merged. Thanks and congratulations with your first contribution :)

**Comment by RodneyLaver** [comment]
Uuuh - thank you again for helping me learn the ropes :)
You are hell of a patient guy and a great teacher!

**Comment by pasis** [comment]
thanks for the kind words ;)
